src/app-bridge/with-authorization.tsx

```diff
@@ -0,0 +1,57 @@
+import { NextPage } from "next";
+import * as React from "react";
+import { PropsWithChildren, ReactNode } from "react";
+
+import { isInIframe, useIsMounted } from "../util";
+import { useDashboardToken } from "./use-dashboard-token";
+
+function SimpleError({ children }: PropsWithChildren<{}>) {
+  return (
+    <div style={{ padding: 32, color: "red" }}>
+      <p>{children}</p>
+    </div>
+  );
+}
+
+type Props = {
+  unmounted?: ReactNode;
+  notIframe?: ReactNode;
+  noDashboardToken?: ReactNode;
+  dashboardTokenInvalid?: ReactNode;
+};
+
+export const withAuthorization =
```

Comment:
Not sure if thats a good message, but would like to have some anyway:

```suggestion
/**
* Render component only if it's requested from the dashboard environment.
**/
export const withAuthorization =
```